documents/wg-application.md

```diff
@@ -0,0 +1,93 @@
+# CUDA WG - draft
+
+## What value do you want to bring to the project?
+
+The Working Group is an attempt to combine efforts to bring reliable support of writing safe GPGPU CUDA code with Rust.
+The work that will be done is supposed to clear the path for other GPGPU platforms in future.
+
+## Why should it be put into place?
+
+First steps to basic support of the feature have already been made by individual developers.
+Regardless of recent progress, there are still a lot of questions about safety and soundness of SIMT (Single Instruction, Multiple Threads) code and they require a lot of collaboration to find solutions.
+
+## What is the working group about?
+
+We want to work together on getting a solid foundation for writing CUDA code.
+Getting a safe and production-ready development experience on Stable Rust is our primary goal!
+
+Also, currently a major obstacle for developing GPGPU applications and algorithms in Rust is a lack of learning resources.
+We plan to solve the "documentation debt" with a broad range of tutorials, examples and references.
+
+## What is the working group not about?
+
+The WG is not focused on promoting or developing "standard" frameworks.
+Instead, we want to provide basic and reliable support of the feature and inspire the community to start using it.
+This should lead to experimenting with different approaches on how to use it and creating awesome tooling.
+
+## Is your WG long-running or temporary?
+
+In our current vision, the WG should live until we fulfill our goals.
+
+In the end, we hope the WG will evolve into another one to cover similar topics:
+to support other GPGPU platforms or to create higher-level frameworks to improve end-to-end experience based on community feedback.
+
+## What is your long-term vision?
+
+Having a reliable and safe CUDA development experience is our ultimate goal.
+This should include:
+
+* Getting `nvptx64-nvidia-cuda` to a [Tier 2](https://forge.rust-lang.org/platform-support.html) support state.
+* Test infrastructure on a real hardware and running the tests during Rust CI process.
+* Rich documentation, references, tutorials and examples.
+* A broad set of new compile errors, warnings and lints for SIMT execution model to avoid pitfalls and ensure code soundness.
+
+## How do you expect the relationship to the project be?
+
+The WG will be responsible for CUDA related issues and user requests.
```

Comment:
I don't think we should be responsible for this, people can request all sort of things. I'd just document here what our current relationship with the project is. For example, we send PRs to Rust components, pick up Alex Crichton, which would be our Core Team Liasson, as a reviewer, etc.

Reply:
Oh, that was a bad wording :smiley:  I rather meant that we will participate in discussions around everything that's related to NVPTX target.

Reply:
Don't think we should mention somebody particular at least at this point. It might feel like we *are obligating* the person with helping us.

Reply:
IIRC, they volunteered back then to play the role of the core team member responsible for the WG, but I know they have different priorities right now, so I'd ask them again. 

From the POV of view of reviewing PRs to rust components, they are probable the person having the most complete picture of target-support/target-specific stuff. So picking Alex as a reviewer is probably the right thing to do anyways. If they don't have time to review they'll say so, but they will at least be able to suggest who should review it instead.

Reply:
Oh, nice, I didn't know about this.

> So picking Alex as a reviewer is probably the right thing to do anyways.

Totally agree here!

Still, do you think we should mention the fact in the application? Seems a little too much going into details, for me.